provider/storage_benchmark.py

```diff
@@ -0,0 +1,301 @@
+                return True
+        logging.info('Could not to get version. %s' % results)
+        return False
+
+    def setup(self, **kwargs):
+        """
+        Download iozone and install it in guest.
+        """
+        raise NotImplementedError
+
+    def post_result(self, results_path, analysis_dir):
+        """
+        Post result after iozone test, generate a series graphs.
+
+        :param results_path: results path.
+        :param analysis_dir: analyzed directory.
+        """
+        # use this function, params "-i" for iozone is not suggested.
+        # suggest usage:iozone.exe -azR -r 64k -n 125M -g 512M -M -f
+        # I:\testfile". Otherwise, your result can't be analysed correctly.
+        logging.info("Generate graph of test result.")
+        iozone_analyzer = postprocess_iozone.IOzoneAnalyzer(
+            list_files=[results_path], output_dir=analysis_dir)
+        iozone_analyzer.analyze()
+        iozone_plotter = postprocess_iozone.IOzonePlotter(
+            results_file=results_path, output_dir=analysis_dir)
+        iozone_plotter.plot_all()
+
+    def run(self, cmd_options='-a', record=True, post=True,
+            timeout=1800, debug=True):
+        """
+        Run iozone test in guest.
+
+        :param cmd_options: iozone command options.
+        e.g: -azR -r 64k -n 1G -g 1G -M -b iozone.xls -f E:\testfile.
+        :param record: bool, if is True, will record the test results,
+        otherwise do not.
+        :param post: bool, if is True, will post the test results,
+        otherwise do not.
+        :param timeout: timeout
+        :param debug: bool, if is True, print the output of testing.
+        """
+        if 'iozone' in cmd_options.split()[0]:
+            self.test.error('The argument cmd_options are '
+                            'not command line options')
+
+        random_string = utils_misc.generate_random_string(8)
+        stamp = time.strftime('%Y-%m-%d-%H-%M-%S') + random_string
+        self.session_check()
+
+        cmd = ' '.join((self.tool_path, cmd_options))
+        status, output = self.session.cmd_status_output(cmd, timeout=timeout)
+        if debug:
+            logging.debug(output)
+
+        if record:
+            results_name = ''.join(('iozone-test-', stamp, '.log'))
+            results_path = record_test_result(self.test, output, results_name)
+
+        analysis_name = 'iozone-analysis-results-' + stamp
+        analysis_path = os.path.join(self.test.resultsdir, analysis_name)
+        if post and record:
+            if not re.search(r'\s+-i\s+\S+', cmd_options):
+                self.post_result(results_path, analysis_path)
+            else:
+                logging.warn('Maybe not be analysed correctly '
+                             'since iozone commands includes \'-i\' '
+                             'option. So Cancel it')
+        if status:
+            self.test.error('Failed to execute %s. %s' % (cmd, output))
```

Comment:
same with before, suggest add '\n' to print output message from new line. and same advice for others need print output.
like self.test.error('Failed to execute %s.\n %s' % (cmd, output))